web/html/src/components/hub/HierarchicalChannelsTable.tsx

```diff
@@ -0,0 +1,213 @@
+    // Build lookup map by channel label
+    const channelMap: Record<string, number> = {};
+    channels.forEach((channel, index) => {
+      channelMap[channel.channelLabel] = index;
+    });
+    // Create copy with added hierarchical properties
+    return channels.map((channel) => {
+      const parentId =
+        channel.parentChannelLabel && channelMap[channel.parentChannelLabel] !== undefined
+          ? channels[channelMap[channel.parentChannelLabel]].channelId
+          : null;
+      // Create a hierarchical row that satisfies both Channel and HierarchicalRow types
+      return {
+        ...channel,
+        id: channel.channelId,
+        parentId,
+      };
+    });
+  }, [channels]);
+
+  // Filter data based on search criteria and architecture filter
+  const filteredData = useMemo(() => {
+    return hierarchicalData.filter((channel) => {
+      // Apply architecture filter
+      if (selectedArchs.length > 0 && !selectedArchs.includes(channel.channelArch)) {
+        return false;
+      }
+      // Apply search criteria
+      if (searchCriteria) {
+        const searchTerm = searchCriteria.toLowerCase();
+        return channel.channelLabel.toLowerCase().includes(searchTerm);
+      }
+      return true;
+    });
+  }, [hierarchicalData, searchCriteria, selectedArchs]);
+
+  // Handle search change
+  const handleSearchChange = useCallback((criteria: string) => {
+    setSearchCriteria(criteria);
+  }, []);
+
+  // Identifier function for the hierarchical table
+  const identifier = useCallback((row: HierarchicalRow): string | number => {
+    return row.id;
+  }, []);
+
+  // Handle architecture filter changes
+  const handleArchFilterChange = useCallback((_, selectedOptions: any) => {
+    const selectedValues = Array.isArray(selectedOptions) ? selectedOptions.map((option) => option.value) : [];
+    setSelectedArchs(selectedValues);
+  }, []);
+
+  // Row class based on sync status and change tracking
+  const rowClass = useCallback((row: any) => {
+    const channel = row as ChannelWithHierarchy;
+    const isCurrentlySynced = channel.synced;
+    let className = isCurrentlySynced ? "synced-channel" : "";
+    return className;
+  }, []);
+
+  // Render the sync checkbox
+  const renderSyncCell = useCallback((row: ChannelWithHierarchy) => {
+    const channelId = row.channelId;
+    const isCurrentlySynced = row.synced;
+    const markedForOperation = row.markedForOperation;
+    // The checkbox should be checked if:
+    // 1. The channel is currently synced AND NOT marked for operation, OR
+    // 2. The channel is NOT currently synced BUT marked for operation
+    const checked = (isCurrentlySynced && !markedForOperation) || (!isCurrentlySynced && markedForOperation);
+    return (
+      <div className="sync-checkbox-container">
+        <input type="checkbox" checked={checked} onChange={(e) => onChannelSelect(channelId, e.target.checked)} />
+      </div>
+    );
+  }, []);
+
+  // Render channel label cell
```

Comment:
Here and elsewhere, commentary like this is not useful, the comment is just the function name with whitespace.